File: backend/microservice/src/level/api/internal/admin/AssignSlotInternalApi.scala
package microservice.level.api.internal.admin

import cats.effect.IO
import java.sql.Connection
import microservice.infrastructure.api.{APIMessage, PlanSteps}
import microservice.infrastructure.http.HttpError
import microservice.level.objects.inventory.BirdPool
import microservice.level.objects.slot.SlotAssignment
import microservice.level.support.admin.SlotAssignmentSupport

/** 模块间 API：分配关卡槽位；由 admin HTTP API 调用，不挂路由。 */
final case class AssignSlotInternalAPIMessage(
  levelSuffix: String,
  submissionId: String,
  assignedById: String,
  note: Option[String],
  birdPool: BirdPool
) extends APIMessage[SlotAssignment] {
  override def plan(connection: Connection): IO[Either[HttpError, SlotAssignment]] =
    PlanSteps.finish {
      SlotAssignmentSupport.requireAssign(connection, levelSuffix, submissionId, assignedById, note, birdPool)
    }
}

File: frontend/src/api/level/internal/admin/assignment/AssignSlotInternalApi.ts
import { z } from "zod";
import { APIMessage } from "../../../../../system/api/APIMessage.js";

/** Mirrors backend internal API level/api/internal/admin/assignment/AssignSlotInternalApi.scala; not registered as a public frontend route. */
export class AssignSlotInternalAPI extends APIMessage<unknown> {
  readonly levelSuffix: unknown;
  readonly submissionId: unknown;
  readonly assignedById: unknown;
  readonly note: unknown;
  readonly birdPool: unknown;
  constructor(levelSuffix: unknown, submissionId: unknown, assignedById: unknown, note: unknown, birdPool: unknown) {
    super();
    this.levelSuffix = levelSuffix;
    this.submissionId = submissionId;
    this.assignedById = assignedById;
    this.note = note;
    this.birdPool = birdPool;
  }

  override get responseSchema() {
    return z.unknown();
  }
}
